### PR TITLE
feat(config): make check for frontend instead of gms more robust

### PR DESCRIPTION
**Pull request by anshbansal**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/src/datahub/emitter/rest_emitter.py [hunk 1]
```diff
@@ -72,10 +72,6 @@ def __init__(
         extra_headers: Optional[Dict[str, str]] = None,
         ca_certificate_path: Optional[str] = None,
     ):
-        if ":9002" in gms_server:
-            logger.warning(
-                "the rest emitter should connect to GMS (usually port 8080) instead of frontend"
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/emitter/rest_emitter.py:77]: Not necessarily - going forward, datahub-frontend can serve as a proxy to GMS. People can point to datahub-frontend instead to minimize the access points they need to maintain.

File: metadata-ingestion/src/datahub/emitter/rest_emitter.py [hunk 2]
```diff
@@ -129,6 +125,19 @@ def test_connection(self) -> None:
...
+        if ":9002" in self._gms_server:
+            logger.warning(
+                "the rest emitter should connect to GMS (usually port 8080) instead of frontend"
+            )
+        if "config" in config:
+            config_obj = config["config"]
+            if "application" in config_obj:
+                application = config_obj["application"]
+                raise ValueError(
+                    f"You should be connecting to GMS URL not {application}"
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/emitter/rest_emitter.py:137]: This does not need to be so strict

File: metadata-ingestion/src/datahub/emitter/rest_emitter.py [hunk 3]
```diff
@@ -129,6 +125,19 @@ def test_connection(self) -> None:
         response = self._session.get(f"{self._gms_server}/config")
         response.raise_for_status()
         config: dict = response.json()
+        if ":9002" in self._gms_server:
+            logger.warning(
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/emitter/rest_emitter.py:129]: Not necessarily - going forward, datahub-frontend can serve as a proxy to GMS. People can point to datahub-frontend instead to minimize the access points they need to maintain.

**Review by shirshanka — APPROVED** [review]
LGTM. Will wait for CI before merging.

**Pull request merged**